pypsrp/__init__.py

```diff
@@ -31,6 +31,11 @@ def _setup_logging(logger):
 logger = logging.getLogger(__name__)
 _setup_logging(logger)
 
+# Silence used libraries
+logging.getLogger("requests").setLevel(logging.WARNING)
```

Comment:
Why is this set here? It should be up to the end user and not this library as to whether these loggers are active or not.

Reply:
If you copy a file, you'll end up with thousands of "POST /wsman HTTP/1.1" in your log, with no real help to the end-user. If the user would have performed a request himself there may be value in knowing what was done. And a "200 OK" was returned and X bytes where send.

But here the user is using WinRM and seeing a handful of HTTP requests brings no value. There is no real relation unless you are troubleshooting the connection, maybe.

Reply:
We could move this to the psrp connection plugin, but IMO anyone using the library shouldn't be getting these unless maybe they are troubleshooting.

Reply:
What I'm trying to say is that if you have enabled logging for all packages then yes you would have thousands of post entries but logging is meant to be configurable by the end user. This library should not be making the decision as to whether or not these libraries add to the log entry.

Reply:
So you basically are saying, if people use my library (through Ansible) it is fine if their log-files are filled with useless HTTP-messages that bear no resemblance to what they are doing.

The only thing an end-user can do currently is enable or disable *all* messages from the use of a library anywhere in Ansible. They can't filter out useless requests-message coming from PSRP, and again there is absolutely no use in having these messages as an end-user, i.e. WinRM does not do this. None if the connection-plugins do this.

Reply:
And it's not unusual (from a connection plugin), still I think being the PSRP library there's absolutely no use in seeing the "informational" but nevertheless useless "POST /wsman HTTP/1.1" messages. There is absolutely no value in that.

```
[dag@moria ansible.git]$ grep -rE '(logging.getLogger|setLevel\()' lib/ansible/plugins
lib/ansible/plugins/callback/logdna.py:            self.log = logging.getLogger('logdna')
lib/ansible/plugins/callback/logdna.py:            self.log.setLevel(logging.INFO)
lib/ansible/plugins/callback/syslog_json.py:        self.logger = logging.getLogger('ansible logger')
lib/ansible/plugins/callback/syslog_json.py:        self.logger.setLevel(logging.DEBUG)
lib/ansible/plugins/callback/logstash.py:            self.logger = logging.getLogger('python-logstash-logger')
lib/ansible/plugins/callback/logstash.py:            self.logger.setLevel(logging.DEBUG)
lib/ansible/plugins/connection/network_cli.py:            logging.getLogger('paramiko').setLevel(logging.DEBUG)
lib/ansible/plugins/connection/netconf.py:logging.getLogger('ncclient').setLevel(logging.INFO)
```

Reply:
I'm failing to see where the issue actually lies, yes it's true pywinrm has no logging but if I understand the issue correctly then requests messages would still be created? If that is not the case then either the logging implementation in pypsrp is wrong or something else is going on.

For example if I create the following log file

```
{
    "version": 1,
    "disable_existing_loggers": false,
    "formatters": {
        "simple": {
            "format": "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
        }
    },

    "handlers": {
        "file": {
            "class": "logging.handlers.RotatingFileHandler",
            "level": "INFO",
            "formatter": "simple",
            "filename": "pypsrp.log"
        }
    },

    "loggers": {
        "pypsrp": {
            "level": "INFO",
            "handlers": ["file"],
            "propagate": "no"
        }
    }
}
```

And then run Ansible (or a Python script that uses pypsrp) with the environment variable `PYPSRP_LOG_CFG=/path/to/file/above.json`. The file `pypsrp.log` would only contain `INFO` entries that come under the `pypsrp*` namespace. Any other libraries like requests are not in this log. I can change that by changing the values under `logger` to include the other libraries or just `*` but as I am saying this is entirely up to the end user to decide what they want.

I think I need to play around with Ansible and see the exact thing you are seeing to really understand what's going on. Can you give me the details on what you are doing to see all these message.

As I am saying I agree there are some `INFO` entries that probably belong in `DEBUG` but that's separate this this conversation.

Reply:
Example of one task (with pypsrp output removed):
```
2018-10-26 02:19:30,815 p=19053 u=dwieers |  Using module file /home/dwieers/ansible/lib/ansible/modules/windows/win_shell.ps1
2018-10-26 02:19:30,882 urllib3.connectionpool Starting new HTTPS connection (1): 38.38.51.3:5986
2018-10-26 02:19:30,899 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:19:30,900 requests_credssp.credssp Starting TLS handshake process
2018-10-26 02:19:30,901 requests_credssp.credssp Step 1. TLS Handshake, returning token: 16030100f5010000f1030368f43f8a364da523feeaacdf72c823
2c8801dbd62bcf19cb5599f5b683ffeea4000082c02cc03000a3009fcca9cca8ccaac0afc0adc0a3c09fc02bc02f00a2009ec0aec0acc0a2c09ec024c028006b006ac073c0770
0c400c3c023c02700670040c072c07600be00bdc00ac0140039003800880087c009c01300330032009a009900450044009dc0a1c09d009cc0a0c09c003d00c0003c00ba003500
84002f0096004100ff01000046000b000403000102000a000a0008001d001700190018002300000016000000170000000d0020001e06010602060305010502050304010402040
3030103020303020102020203
2018-10-26 02:19:30,901 urllib3.connectionpool Resetting dropped connection: 38.38.51.3
2018-10-26 02:19:30,919 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:19:30,920 requests_credssp.credssp Step 1. TLS Handshake, received token: 16030303090200005103035bd25d93b38b72cb10cd8a2b9a604a5
7a3698ea3ba698fcebe444676241011f02073310000e43e1ed0a8db9c74c07b2ff87e2e0e76312909a179e604c6ecdacc72c02800000900170000ff010001000b0001df0001dc
0001d9308201d53082013ea003020102021069347b861d81638346563a86144d466a300d06092a864886f70d01010505003029312730250603550403131e57534d414e2d73637
66d6d30312e706f643531766d6d2e6163692e6c6162301e170d3138313032353232353734355a170d3139303432333232353734355a3029312730250603550403131e57534d41
4e2d7363766d6d30312e706f643531766d6d2e6163692e6c616230819f300d06092a864886f70d010101050003818d00308189028181009a1b550cd82f27c1eccfceef4e02526
fc86366148b5b5f0b29fdd76fd5b170adf537c8b1cfca205b6690038e9f3e2d5123fc32d7fad80c0c10c9e1359ed1eb55fc0eb496c65b471902ffbf0182411e0d579fe95f73e0
7f38f948067658d5167dc854d6b0470961a296a8fa2eb106ffdf8d8c0404ea1f4133019c1471e9eff4b70203010001300d06092a864886f70d01010505000381810044218ae56
9f0397c4d65206bbb6ec55a4beccc8d286540f6b3cd0ac1683013d60aff9b04bc65071e27f3beacfad45cb10c7ce797e0e9aae06ae819c60ec549d48b3343a30c7aa65c2db1bf
e2d8c4522e5106b745561b2dfd408301d6cb7c4d9a209635afe5a60af79ada35c6cd4f7d8d91ae277466873ef96adccd13490a515b0c0000c90300174104818350aeeaeba1d71
5873792b7a52ee6b92f17958ef0a63638bf59c6f3d75320db1c58813f095c08ef601128cbd0f540d318638fccf63ccbca67175bf26b4f8f0201008030eb1f6fdaf0e81c84b168
71f6aacbe334d7b3fa3590947b553407a2e2e21b9d6935035460a989b015bb42a09f2538911083cbf75d5bccf256395a2ae67a4639d357fe39525991516683b4fc4480ae73213
770a4de4702fa74d30545204d44aa877985de205215027a05f8a512ffae9d1d97018f69e3f675adfd3ef983f89d9b0e000000
2018-10-26 02:19:30,921 requests_credssp.credssp Step 1. TLS Handshake, returning token: 16030300461000004241044a74d32d689420bac39c0c11309a63
941cb866078458a6f7ad87476cd8177d103a2cfb3272632ee9a9c4859fc0bc5e87116c544f611a8034fe39c2b088f4b20514030300010116030300602e31f5dd2490d2391ff1a
82ddc9ab08c307d1daa703ee172cd4bd330bd45cf0a64696a01ad0f61ea5556dfa75f06cece0118900c03174a4cfd6f8d6898b9ff82c5606e70df6643d3ee76b9acc4f950ed86
3276b7dbfb933902b18ef242b9f2e2
2018-10-26 02:19:30,923 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:19:30,924 requests_credssp.credssp Step 1. TLS Handshake, received token: 14030300010116030300606ff086dfffb590176411ccca3cd1b4975ad21ada9333fc29166d6ece4ae56de3966a19f97f22feca409f90c249209f0694b3da0f8596ff53f8d40afe1e3fccc08d4809ca424ceb8f238ffca07e26574042132da649d001548689aed14f21a990
2018-10-26 02:19:30,924 requests_credssp.credssp TLS Handshake complete. Protocol: TLSv1.2, Cipher: ECDHE-RSA-AES256-SHA384
2018-10-26 02:19:30,924 requests_credssp.credssp Starting Authentication process
2018-10-26 02:19:30,924 requests_credssp.spnego SSPI or GSSAPI is not available, using ntlm-auth as auth backend
2018-10-26 02:19:30,925 requests_credssp.spnego NTLM Negotiate message: 4e544c4d5353500001000000329088e2000000002800000000000000280000000601b11d0000000f
2018-10-26 02:19:30,925 requests_credssp.credssp Step 2. Authenticate, returning token: 3037a003020106a130302e302ca02a04284e544c4d5353500001000000329088e2000000002800000000000000280000000601b11d0000000f
2018-10-26 02:19:30,927 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:19:30,927 requests_credssp.credssp Step 3. Authenticate, received token: 30820106a003020103a181fe3081fb3081f8a081f50481f24e544c4d53535000020000000800080038000000368289e2851d3352452a4a2a0000000000000000b200b20040000000060380250000000f504f443531564d4d0200100070006f0064003500310076006d006d0001000e005300430056004d004d00300031000400200070006f0064003500310076006d006d002e006100630069002e006c0061006200030030007300630076006d006d00300031002e0070006f0064003500310076006d006d002e006100630069002e006c00610062000500200070006f0064003500310076006d006d002e006100630069002e006c0061006200070008002b669590c16cd40100000000
2018-10-26 02:19:30,928 requests_credssp.spnego NTLM: Parsing Challenge message: 4e544c4d53535000020000000800080038000000368289e2851d3352452a4a2a0000000000000000b200b20040000000060380250000000f504f443531564d4d0200100070006f0064003500310076006d006d0001000e005300430056004d004d00300031000400200070006f0064003500310076006d006d002e006100630069002e006c0061006200030030007300630076006d006d00300031002e0070006f0064003500310076006d006d002e006100630069002e006c00610062000500200070006f0064003500310076006d006d002e006100630069002e006c0061006200070008002b669590c16cd40100000000
2018-10-26 02:19:30,928 requests_credssp.spnego NTLM: Generating Authenticate message
2018-10-26 02:19:30,930 requests_credssp.credssp Starting public key verification process at version 3
2018-10-26 02:19:30,930 requests_credssp.credssp Reported server version was 3, susceptible to MitM attacks and should be patched - CVE 2018-0886
2018-10-26 02:19:30,931 requests_credssp.credssp Step 3. Server Authentication, returning token: 30820232a003020106a1820187308201833082017fa082017b048201774e544c4d53535000030000001800180065000000ea00ea007d00000000000000580000000d000d005800000000000000650000001000100067010000368289e20601b11d0000000f3a63c2c2e49e7eeb6680247100e334b541646d696e6973747261746f720000000000000000000000000000000000000000000000004694f45704100b87f60c4b1b65990b2101010000000000002b669590c16cd401990508e625c1faa0000000000200100070006f0064003500310076006d006d0001000e005300430056004d004d00300031000400200070006f0064003500310076006d006d002e006100630069002e006c0061006200030030007300630076006d006d00300031002e0070006f0064003500310076006d006d002e006100630069002e006c00610062000500200070006f0064003500310076006d006d002e006100630069002e006c0061006200070008002b669590c16cd40106000400020000000000000000000000ff657cfb414b11447b719a2e639b1d72a3819f04819c010000000a2e2a014e31b69f000000009caa6ac829f4b1bce49bb1fad1e657e126acef4091323cf2658c7619a046d5f177f31ce1c4e646a87503f0336a3a843650baa432161bbebcc70eb64a6a1eb38853a51af7a6b02c3482110e641f2ec4d816712b9a3ef33b0e6cc9319b89cb27780bec7f5cbfc0cabf8e7554a489589e107dc6501f1e14c535efd61ab78cba56949ff7bdd6dbb2a7779dced249
2018-10-26 02:19:30,933 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:19:30,933 requests_credssp.credssp Step 3. Server Authentication, received token: 3081a7a003020103a3819f04819c01000000ed5292b489c93ca000000000393554243fb54050d313b685909812ee2d3ae8917f7d49d9df83e866ac7bb267cd5aebcec3f6d99670cdfd284f5e3e874f0adde08284d56e8a4c6292eba642c016bcb4f9617c7874a544b1d3e0ac7d5ff892265d00301d02ed08437ba0cb14bd11274575031fcf8371eeaea906ab3f170fc29469ab1bebde2b7c1370ddcc7919d5c6737a295ffabb13be5288
2018-10-26 02:19:30,933 requests_credssp.credssp Starting server public key response verification
2018-10-26 02:19:30,934 requests_credssp.credssp Sending encrypted credentials
2018-10-26 02:19:31,028 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 1991
2018-10-26 02:19:31,174 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 2797
2018-10-26 02:19:31,187 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 930
2018-10-26 02:19:31,213 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 847
2018-10-26 02:19:31,236 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:19:51,347 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 500 1318
2018-10-26 02:19:51,544 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 6205
2018-10-26 02:19:51,558 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 602
2018-10-26 02:19:51,563 p=19053 u=dwieers |  TASK [Add ISO to library] ******************************************************
2018-10-26 02:19:51,563 p=19053 u=dwieers |  task path: /home/dwieers/backend/task073-2012r2_scvmm_step03.yaml:473
2018-10-26 02:19:51,564 p=19053 u=dwieers |  changed: [bdsol-aci51-scvmm-01] => {
    "changed": true,
    "cmd": "Import-SCLibraryPhysicalResource -VMMServer 38.38.51.3 -SourcePath \"C:\\Windows\\Temp\\RCDx64_13_03_12.iso\" -SharePath \"\\\\scvmm01.pod51vmm.aci.lab\\MSSCVMMLibrary\" -OverwriteExistingFiles",
    "delta": "0:00:18.204877",
    "end": "2018-10-26 12:19:51.580968",
    "rc": 0,
    "start": "2018-10-26 12:19:33.376091",
    "stderr": "",
    "stderr_lines": [],
    "stdout": "\r\n\r\nRelease               : \r\nState                 : Normal\r\nLibraryShareId        : 47e26463-8a62-4eed-ab52-a9a3d8a2dbc8\r\nSharePath             : \\\\scvmm01.pod51vmm.aci.lab\\MSSCVMMLibrary\\RCDx64_13_03_\r\n                        12.iso\r\nFileShare             : \r\nDirectory             : \\\\scvmm01.pod51vmm.aci.lab\\MSSCVMMLibrary\r\nSize                  : 108822528\r\nIsOrphaned            : False\r\nFamilyName            : \r\nNamespace             : Global\r\nReleaseTime           : \r\nHostVolumeId          : d689cf85-a922-4181-8625-d346af5d7df0\r\nHostVolume            : \r\nClassification        : \r\nHostId                : 3be0b6d7-40fb-4790-8fc9-1240867ed2d0\r\nHostType              : LibraryServer\r\nHostName              : scvmm01.pod51vmm.aci.lab\r\nVMHost                : \r\nLibraryServer         : scvmm01.pod51vmm.aci.lab\r\nCloudId               : \r\nCloud                 : \r\nLibraryGroup          : \r\nGrantedToList         : {}\r\nUserRoleID            : 00000000-0000-0000-0000-000000000000\r\nUserRole              : \r\nOwner                 : \r\nObjectType            : ISO\r\nAccessibility         : Public\r\nName                  : RCDx64_13_03_12.iso\r\nIsViewOnly            : False\r\nDescription           : \r\nAddedTime             : 10/26/2018 2:19:49 AM\r\nModifiedTime          : 10/26/2018 2:19:49 AM\r\nEnabled               : True\r\nMostRecentTask        : \r\nServerConnection      : Microsoft.SystemCenter.VirtualMachineManager.Remoting.S\r\n                        erverConnection\r\nID                    : d86d9059-4bcd-4ee9-91ec-576b03d7894b\r\nMarkedForDeletion     : False\r\nIsFullyCached         : True\r\nMostRecentTaskIfLocal : \r\n\r\n\r\n\r\n",
    "stdout_lines": [
        "",
        "",
        "Release               : ",
        "State                 : Normal",
        "LibraryShareId        : 47e26463-8a62-4eed-ab52-a9a3d8a2dbc8",
        "SharePath             : \\\\scvmm01.pod51vmm.aci.lab\\MSSCVMMLibrary\\RCDx64_13_03_",
        "                        12.iso",
        "FileShare             : ",
        "Directory             : \\\\scvmm01.pod51vmm.aci.lab\\MSSCVMMLibrary",
        "Size                  : 108822528",
        "IsOrphaned            : False",
        "FamilyName            : ",
        "Namespace             : Global",
        "ReleaseTime           : ",
        "HostVolumeId          : d689cf85-a922-4181-8625-d346af5d7df0",
        "HostVolume            : ",
        "Classification        : ",
        "HostId                : 3be0b6d7-40fb-4790-8fc9-1240867ed2d0",
        "HostType              : LibraryServer",
        "HostName              : scvmm01.pod51vmm.aci.lab",
        "VMHost                : ",
        "LibraryServer         : scvmm01.pod51vmm.aci.lab",
        "CloudId               : ",
        "Cloud                 : ",
        "LibraryGroup          : ",
        "GrantedToList         : {}",
        "UserRoleID            : 00000000-0000-0000-0000-000000000000",
        "UserRole              : ",
        "Owner                 : ",
        "ObjectType            : ISO",
        "Accessibility         : Public",
        "Name                  : RCDx64_13_03_12.iso",
        "IsViewOnly            : False",
        "Description           : ",
        "AddedTime             : 10/26/2018 2:19:49 AM",
        "ModifiedTime          : 10/26/2018 2:19:49 AM",
        "Enabled               : True",
        "MostRecentTask        : ",
        "ServerConnection      : Microsoft.SystemCenter.VirtualMachineManager.Remoting.S",
        "                        erverConnection",
        "ID                    : d86d9059-4bcd-4ee9-91ec-576b03d7894b",
        "MarkedForDeletion     : False",
        "IsFullyCached         : True",
        "MostRecentTaskIfLocal : ",
        "",
        "",
        ""
    ]
}
```
Example of using win_copy:
```
2018-10-26 02:16:36,917 p=19053 u=dwieers |  Using module file /home/dwieers/ansible/lib/ansible/modules/windows/win_copy.ps1
2018-10-26 02:16:36,984 urllib3.connectionpool Starting new HTTPS connection (1): 38.38.51.3:5986
2018-10-26 02:16:37,005 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:16:37,007 requests_credssp.credssp Starting TLS handshake process
2018-10-26 02:16:37,007 requests_credssp.credssp Step 1. TLS Handshake, returning token: 16030100f5010000f10303a900844826dd77dddb507d996e0c7f
c22791a829f81901842145d6e0dd60c757000082c02cc03000a3009fcca9cca8ccaac0afc0adc0a3c09fc02bc02f00a2009ec0aec0acc0a2c09ec024c028006b006ac073c0770
0c400c3c023c02700670040c072c07600be00bdc00ac0140039003800880087c009c01300330032009a009900450044009dc0a1c09d009cc0a0c09c003d00c0003c00ba003500
84002f0096004100ff01000046000b000403000102000a000a0008001d001700190018002300000016000000170000000d0020001e060106020603050105020503040104020403030103020303020102020203
2018-10-26 02:16:37,009 urllib3.connectionpool Resetting dropped connection: 38.38.51.3
2018-10-26 02:16:37,028 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:16:37,029 requests_credssp.credssp Step 1. TLS Handshake, received token: 16030303090200005103035bd25ce5094a24c6fc0f3b6a540906de3fc37868e6c467666ed876c767c44888201b3d00001fe116cb0f8c6c782f32496e54d509c71ce5f046aa06d10ea31127b6c02800000900170000ff010001000b0001df0001dc0001d9308201d53082013ea003020102021069347b861d81638346563a86144d466a300d06092a864886f70d01010505003029312730250603550403131e57534d414e2d7363766d6d30312e706f643531766d6d2e6163692e6c6162301e170d3138313032353232353734355a170d3139303432333232353734355a3029312730250603550403131e57534d414e2d7363766d6d30312e706f643531766d6d2e6163692e6c616230819f300d06092a864886f70d010101050003818d00308189028181009a1b550cd82f27c1eccfceef4e02526fc86366148b5b5f0b29fdd76fd5b170adf537c8b1cfca205b6690038e9f3e2d5123fc32d7fad80c0c10c9e1359ed1eb55fc0eb496c65b471902ffbf0182411e0d579fe95f73e07f38f948067658d5167dc854d6b0470961a296a8fa2eb106ffdf8d8c0404ea1f4133019c1471e9eff4b70203010001300d06092a864886f70d01010505000381810044218ae569f0397c4d65206bbb6ec55a4beccc8d286540f6b3cd0ac1683013d60aff9b04bc65071e27f3beacfad45cb10c7ce797e0e9aae06ae819c60ec549d48b3343a30c7aa65c2db1bfe2d8c4522e5106b745561b2dfd408301d6cb7c4d9a209635afe5a60af79ada35c6cd4f7d8d91ae277466873ef96adccd13490a515b0c0000c90300174104818350aeeaeba1d715873792b7a52ee6b92f17958ef0a63638bf59c6f3d75320db1c58813f095c08ef601128cbd0f540d318638fccf63ccbca67175bf26b4f8f0201008036962a1333fbada7be1126ec2dbc745a47acba1367ecc53affabec994f9e8f8d15529162bd687b0212c2362c126646f00ef94ba8aadc6a8ca5def93fe887a57a87403cc2f2e78364450ea696b42b811cc4fd50a795a380b3c9366812da6be1d35ee49adf0688d7ce60afd5a0a8c105149ea08c3d3350e030aba3f262009e419c0e000000
2018-10-26 02:16:37,029 requests_credssp.credssp Step 1. TLS Handshake, returning token: 1603030046100000424104f9c2e00b8919dc03b2a5ba7663cb7789ea0ab13b0d489d134dfa85b98bc5a481a1108273419a8e3fbffead30d505776da48c6c5e98e2a4f911c8159edf79c0051403030001011603030060458ade1170d4bd08399ae80dd75b2d9976cbbbf01650789f8b1e3044bc464e19d06a87f6e52d36b5986ed0ff639bbca135c6ad5a7d6f6cc6a76f8b2a9b6eb0128884365f31abe7ebcdacc1eb6dfabd6127ab36604a0320e4c3348280adfcda9a
2018-10-26 02:16:37,033 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:16:37,033 requests_credssp.credssp Step 1. TLS Handshake, received token: 1403030001011603030060dc8ea6a828e176398614597631419161fe37b8759882bb989358394f8badd68183976a0efe08e03e8a6da320f6ece143bdc6fb86ad3bb8c24a69b8df5a25af6c336ac64a81b45ce06958d67bd3e851c1ecad05b04ab1bca501bcc48d98b5a979
2018-10-26 02:16:37,034 requests_credssp.credssp TLS Handshake complete. Protocol: TLSv1.2, Cipher: ECDHE-RSA-AES256-SHA384
2018-10-26 02:16:37,035 requests_credssp.credssp Starting Authentication process
2018-10-26 02:16:37,035 requests_credssp.spnego SSPI or GSSAPI is not available, using ntlm-auth as auth backend
2018-10-26 02:16:37,035 requests_credssp.spnego NTLM Negotiate message: 4e544c4d5353500001000000329088e2000000002800000000000000280000000601b11d0000000f
2018-10-26 02:16:37,036 requests_credssp.credssp Step 2. Authenticate, returning token: 3037a003020106a130302e302ca02a04284e544c4d5353500001000000329088e2000000002800000000000000280000000601b11d0000000f
2018-10-26 02:16:37,037 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:16:37,038 requests_credssp.credssp Step 3. Authenticate, received token: 30820106a003020103a181fe3081fb3081f8a081f50481f24e544c4d53535000020000000800080038000000368289e267981820d89df3780000000000000000b200b20040000000060380250000000f504f443531564d4d0200100070006f0064003500310076006d006d0001000e005300430056004d004d00300031000400200070006f0064003500310076006d006d002e006100630069002e006c0061006200030030007300630076006d006d00300031002e0070006f0064003500310076006d006d002e006100630069002e006c00610062000500200070006f0064003500310076006d006d002e006100630069002e006c0061006200070008007c04ef28c16cd40100000000
2018-10-26 02:16:37,038 requests_credssp.spnego NTLM: Parsing Challenge message: 4e544c4d53535000020000000800080038000000368289e267981820d89df3780000000000000000b200b20040000000060380250000000f504f443531564d4d0200100070006f0064003500310076006d006d0001000e005300430056004d004d00300031000400200070006f0064003500310076006d006d002e006100630069002e006c0061006200030030007300630076006d006d00300031002e0070006f0064003500310076006d006d002e006100630069002e006c00610062000500200070006f0064003500310076006d006d002e006100630069002e006c0061006200070008007c04ef28c16cd40100000000
2018-10-26 02:16:37,038 requests_credssp.spnego NTLM: Generating Authenticate message
2018-10-26 02:16:37,040 requests_credssp.credssp Starting public key verification process at version 3
2018-10-26 02:16:37,040 requests_credssp.credssp Reported server version was 3, susceptible to MitM attacks and should be patched - CVE 2018-0886
2018-10-26 02:16:37,041 requests_credssp.credssp Step 3. Server Authentication, returning token: 30820232a003020106a1820187308201833082017fa082017b048201774e544c4d53535000030000001800180065000000ea00ea007d00000000000000580000000d000d005800000000000000650000001000100067010000368289e20601b11d0000000f9842ecf4cc54e70a1fe22eb3f270933241646d696e6973747261746f720000000000000000000000000000000000000000000000008205938311e731a6883fb2d53e892ded01010000000000007c04ef28c16cd40198ae440834934a8c000000000200100070006f0064003500310076006d006d0001000e005300430056004d004d00300031000400200070006f0064003500310076006d006d002e006100630069002e006c0061006200030030007300630076006d006d00300031002e0070006f0064003500310076006d006d002e006100630069002e006c00610062000500200070006f0064003500310076006d006d002e006100630069002e006c0061006200070008007c04ef28c16cd4010600040002000000000000000000000032f40d6b256da776522e472e82577b32a3819f04819c010000004737c57b7bc1f3dd000000003f9024da818a57614598a9aa297e801267736ca67eab15d0db862f449dad818aea158becf36f564da862cc7daf62db33063fcc0acb09f084ca5f07b047f51d7b22d384bbb8908c05d5c1b130b125962fef9265d0746367318402889660aec288809d06f42e68421e518684fc34269bd060563783c4f034f9e5cb82e44e9bbc122b4bb3980d87bbbe1182b849
2018-10-26 02:16:37,042 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 401 0
2018-10-26 02:16:37,043 requests_credssp.credssp Step 3. Server Authentication, received token: 3081a7a003020103a3819f04819c01000000f729e0edef2ce89f000000000dc25b05746f88071f0624384fcd9868ccd97272b81ac1acf66eed7c560f6b1a357494a9d330758e899249c54b22f15f124cb0054ddb37ec74af9710893950bf305cab0698a949cadef83f5bce9d880faa32ae4aa813abbcd27a427da0aeb83f58099edbff0214adb5b2a41f5e6b9294257f7b55156aa70ec97804e539aee5f9f13dc003c844cc548e50e03a
2018-10-26 02:16:37,043 requests_credssp.credssp Starting server public key response verification
2018-10-26 02:16:37,044 requests_credssp.credssp Sending encrypted credentials
2018-10-26 02:16:37,138 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 1992
2018-10-26 02:16:37,273 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 2797
2018-10-26 02:16:37,289 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 930
2018-10-26 02:16:37,309 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 847
2018-10-26 02:16:37,772 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 1653
2018-10-26 02:16:37,783 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 847
2018-10-26 02:16:37,798 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 1469
2018-10-26 02:16:37,820 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 1635
2018-10-26 02:16:37,830 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 847
2018-10-26 02:16:37,986 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,000 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,047 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,062 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,071 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,081 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,096 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,111 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,118 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,125 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,133 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,142 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,150 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,158 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,165 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,176 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,185 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,194 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,203 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,212 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,220 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,228 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,236 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,244 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,257 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,265 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,274 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,282 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,293 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,301 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,310 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,320 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,331 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,350 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,362 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,370 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,379 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,387 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,396 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,407 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,415 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,422 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,430 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,437 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,444 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,453 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,460 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,469 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,478 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,486 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,495 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,503 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,511 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,519 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,527 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,535 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,543 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,552 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,560 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,567 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,575 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,584 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,596 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,607 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,615 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,628 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,639 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,647 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,655 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,664 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,672 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,680 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,693 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,703 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,711 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,720 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,728 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,737 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,746 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,755 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,763 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,769 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 753
2018-10-26 02:16:38,880 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 1367
2018-10-26 02:16:38,884 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 757
2018-10-26 02:16:38,888 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 602
2018-10-26 02:16:38,909 p=19053 u=dwieers |  Using module file /home/dwieers/ansible/lib/ansible/modules/windows/win_copy.ps1
2018-10-26 02:16:38,926 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 847
2018-10-26 02:16:39,093 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 1381
2018-10-26 02:16:39,110 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 847
2018-10-26 02:16:39,117 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 1177
2018-10-26 02:16:39,122 urllib3.connectionpool https://38.38.51.3:5986 "POST /wsman HTTP/1.1" 200 602
2018-10-26 02:16:39,127 p=19053 u=dwieers |  TASK [windows-scvmm-cert : Copy APIC-Hyper-V-Agent zip to HyperV server] *******
2018-10-26 02:16:39,127 p=19053 u=dwieers |  task path: /home/dwieers/backend/roles/windows-scvmm-cert/tasks/main.yml:28
2018-10-26 02:16:39,127 p=19053 u=dwieers |  changed: [bdsol-aci51-scvmm-01] => {
    "changed": true,
    "checksum": "254314b77006faca14ad6b6e67fc698a7a8ef080",
    "dest": "C:\\Windows\\Temp\\aci-msft\\aci-msft-pkg-3.1.2o.zip",
    "operation": "file_copy",
    "original_basename": "aci-msft-pkg-3.1.2o.zip",
    "size": 6964318,
    "src": "binaries/aci-msft/aci-msft-pkg-3.1.2o.zip"
}
```

Reply:
Not sure if the requests library is going to change their INFO message, because it's only one message per request. The problem is that pypsrp as a consumer is doing lots of requests for a single action (e.g. copy a file) and each of these messages are basically identical and not giving any useful information (except maybe when debugging).

So what is useful depends largely on its use. IMO someone using pypsrp is not interested in the lower-level requests or requests_credssp information (unless maybe when debugging).

Reply:
Yep it is mostly useless but we shouldn’t be disabling it altogether in case someone really wants to log it. We need to figure out why/where they are enabled and work from there.

Reply:
I am not disabling it altogether. Warnings and errors are still being reported.

Fact is that even if we are not touching it, debug-messages are not being logged either. So regardless it's not a complete stream, things are being filtered out by default too, just a different level.

Reply:
Yep, the work to stop INFO being logged should be done in the place that enabled them in the first place which is Ansible. Whether Ansible logs everything by default is wrong or whether we should filter them further in the psrp connection plugin I'm not sure, but it's definitely not pypsrp place to make this change.

Reply:
That's where we disagree. pypsrp is a consumer of other libraries and is the best place to decide what information from used libraries makes sense to the end-user in this specific context.

The urllib3 HTTP request info is the perfect example of something that may make sense for an application, but in this context is useless in the pypsrp context. The requests_credssp probably needs modifications in the library (I'll make a PR), but if the developer sees things differently, you may want to make the pyprsp output at least more consistent in its log-level output.